### ISSUE TITLE
🐛 `useEnumInitializers` should not emit diagnostics in ambient namespaces

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#19]
### Environment information

```block
main branch.
```

### What happened?

The following code should not emit diagnostics:

```ts
declare namespace X {
  enum Enum {
    A,
    B,
  }
}
```

Here the namespace creates an ambient context, making `Enum` an ambient enum.
The rule thinks that `Enum` is a concrete enum.

Note that the issue is recursive:

```ts
export declare namespace X {
    export namespace Y {
        export enum Enum {
            A,
            B,
        }
    }
}
```

See this example in the [playground](https://biomejs.dev/playground?code=ZQB4AHAAbwByAHQAIABkAGUAYwBsAGEAcgBlACAAbgBhAG0AZQBzAHAAYQBjAGUAIABYACAAewAKACAAIAAgACAAZQB4AHAAbwByAHQAIABuAGEAbQBlAHMAcABhAGMAZQAgAFkAIAB7AAoAIAAgACAAIAAgACAAIAAgAGUAeABwAG8AcgB0ACAAZQBuAHUAbQAgAEUAbgB1AG0AIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAQQAsAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAQgAsAAoAIAAgACAAIAAgACAAIAAgAH0ACgAgACAAIAAgAH0ACgB9AA%3D%3D).

### Expected result

`Enum` should be recognized as an ambient enum and thus the rule should ignore the enum.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
